Update io.js v1.1.0 and node v0.12.0 V8 versions

ruimarinho opened the pull request:
- `io.js@1.1.0 -> V8 4.1.0.14`
- `node@0.12.0 -> V8 3.28.73`

Closes #124.

Commit:
Update io.js 1.1.0 and node 0.12 V8 versions

ruimarinho commented:
@domenic do you have a chance to review this?

domenic commented:
I never understood why were were comparing to a hypothetical future Node 0.12 release anyway. We should just compare to the latest stable, 0.10.35.

Fishrock123 commented:
I agree with domenic.

ruimarinho commented:
Well, I guess it was due to the fact that the last V8 upgrade on the node 0.10 branch was done such a long time ago (one and a half years ago to v3.14.5.9) that it didn't sound like a fair comparison at the time of writing. 0.11.15 is supposed to be the RC version of 0.12, but who knows how many more months will it take for that to happen.

I'm fine with targeting the content for 0.10.35 and it seems like we already have two +1's from Collaborators too, so I'll go ahead and search the internet archives for V8 v3.14 :-).

@domenic, on a side note, I've noticed that 0.11.15 is shipping with V8 3.28.73 which is currently an abandoned version (the branch head of 3.28 is 3.28.71.19). Do you think it's worth to file that in?

domenic commented:
> Do you think it's worth to file that in?

Not sure what "file that in" means?

ruimarinho commented:
To submit an issue to joyent/node's repository.

domenic commented:
Oh. *Shrug*. My cares about joyent/node have plummeted to absolute zero.

ruimarinho commented:
So V8 3.14 does not have any ES6 feature completed, so it seems pointless to compare io.js with node@0.10 with the --harmony since literally no one is using that. However, node@0.11 with the --harmony flag is commonly used (even in production) due to the support for generators that ignited projects and communities like `koa` and `co`.

What do you think about adding a small paragraph about node@0.10's V8 version (for comparison) but keep node@0.11's info without references to 0.12?

domenic commented:
> So V8 3.14 does not have any ES6 feature completed, so it seems pointless to compare io.js with node@0.10 with the --harmony since literally no one is using that.

Agreed. It's more useful to compare io.js to node.js. io.js has ES6 features, and node.js does not.

> What do you think about adding a small paragraph about node@0.10's V8 version (for comparison) but keep node@0.11's info without references to 0.12? 

I might keep something very minimal, without mentioning versions at all. E.g., "If you ever used the unstable Node 0.11.x series with the `--harmony` flag, you'll notice this is very different behavior."

mzgol commented:
So, Node.js 0.12 has been released; it contains v8 `3.28.73` (why is there the 4th number in the version missing?...). This PR should be modified to say `shipped` instead of `will be shipped` and merged.

mikeal commented:
well, Joyent didn't write up anything about the ES6 features now available in the 6 month old version of v8 they shipped so someone would have to manually figure that out in order to write a comparison :(

mzgol commented:
> I'll go ahead and search the internet archives for V8 v3.14

@ruimarinho You probably don't need it now that Node 0.12 has been released but it's `3.14.5.9` (the branch that landed in Chrome 24).

It's easy to check if you use `nvm`. :)
```
$ nvm exec 0.10 node -p process.versions.v8
Running node v0.10.36
3.14.5.9
```

mzgol commented:
> well, Joyent didn't write up anything about the ES6 features now available in the 6 month old version of v8 they shipped so someone would have to manually figure that out in order to write a comparison :(

Yeah, weird. :/ However, as far as I understand it should more or less match Chrome 38.

snostorm commented:
In general we should address 0.12 dropping in a few ways through the product/public relations. That long-awaited release is going to gain a lot of press itself and possibly create some confusion. "Well, io.js is only based on v0.10... but the _new_ Node.js™ is based on v0.11!"

Hmmm, this aspect should be a new issue/discussion. Anyway, re: the version, lets get this PR update or an alternative one to start addressing this specific development.

mzgol commented:
@snostorm Sure. However, what's currently on the site is incorrect and v8 3.28 is still putting most unfinished stuff behind `--harmony`:
```
  --harmony (enable all harmony features (except proxies))
        type: bool  default: false
```
contrary to io.js's v8 version:
```
  --harmony (enable all completed harmony features)
        type: bool  default: false
```
so most of the description *should be* correct. IMO v8 mentions should be updated on the site so that it's what we ended up with and the rest, public relations etc. can be discussed separately.

At least that's my humble opinion. :)

mikeal commented:
Don't worry about the messaging side of it. I'm quite surprised at how *little* press 0.12 is getting at the moment (I have a theory about why that is and it will probably change on Tuesday).

As far as messaging the differences between 0.12 and 0.10 are similar (outdated v8 and libuv). There are no improvements in 0.12 that are not in io.js and node.js is still lagging behind on shipping supported versions of its dependencies.

ruimarinho commented:
I'm updating this now with the content from #124 as well. I was reluctant for more updates as the recent release cycle of 0.11.x was abnormal.

domenic commented on public/es6.html:
```diff
@@ -31,14 +31,14 @@
       <h2 class="faq-title">No more --harmony flag</h2>
       <div class="faq-body">
-        <p>On joyent/node@0.12.x (V8 3.26), the <code>--harmony</code> runtime flag enabled all <strong>completed</strong>, <strong>staged</strong> and <strong>in progress</strong> ES6 features together, in bulk (with the exception of nonstandard/non-harmonious semantics for <code>typeof</code> which were hidden under <code>--harmony-typeof</code>).  This meant that some really buggy or even brok ...
+        <p>On joyent/node@0.12.x (V8 3.28), the <code>--harmony</code> runtime flag enabled all <strong>completed</strong>, <strong>staged</strong> and <strong>in progress</strong> ES6 features together, in bulk (with the exception of <code>proxies</code> which were hidden under <code>--harmony-proxies</code>). This meant that some really buggy or even broken features like <a href="https://develo ...
```
Why remove the note about typeof?

ruimarinho replied:
Correct me if I'm wrong, but I don't think this applies to the V8 version that ships with node 0.12 (3.28) anymore.

Commit:
Replace joyent/node by Node.js™

Commit:
Update node --harmony description to present tense

ruimarinho commented:
Rebased and updated with the new V8 versions for both io.js 1.1.0 and node 0.12.0. I also double-checked that the features enabled via the `--es_staging` flag on io.js continue to be up-to-date (no changes), as well the shipping features (no changes too).

ruimarinho commented:
Interestingly, though, they've shipped 0.12 with an unsupported V8 version, considering the branch head for 3.28 is actually sitting at `3.28.71.19`.

Fishrock123 commented:
landed in https://github.com/iojs/website/commit/44864713784d018aeb606f67ac0fda798083c137 -- thanks!
